src/core/Toast/Toaster.tsx

```diff
@@ -51,22 +38,45 @@ export default class Toaster {
     category: ToastCategory,
     settings?: ToastOptions,
   ) {
-    this.addToast({
-      ...settings,
-      content,
-      category,
-    });
+    const lastId = this.toasts[0]?.id ?? 0;
+    const currentId = lastId + 1;
+    this.toasts = [
+      {
+        ...settings,
+        content,
+        category,
+        onRemove: () => {
+          this.removeToast(currentId);
+          settings?.onRemove;
+        },
+        id: currentId,
+        isVisible: true,
+      },
+      ...this.toasts,
+    ];
+
+    this.updateView();
   }
 
-  public closeAll(): void {
-    this.closeAllHandler();
+  private removeToast(id: number) {
+    this.toasts = this.toasts.filter((toast) => toast.id !== id);
+    this.updateView();
   }
 
-  private _addHandler(addHandler: (settings: ToastSettings) => void) {
-    this.addToast = addHandler;
+  private updateView() {
+    ReactDOM.render(
+      <ToastMaster toasts={this.toasts} />,
+      getContainer(TOASTS_CONTAINER_ID),
```

Comment:
So instead of constructor, `document` is being used `updateView`. Have you checked that this actually solves the SSR problem?

Reply:
I havent. I guess we will release a dev version and test in real project. The idea is to run it when it is needed, not before..